sklearn/decomposition/_nmf.py

```diff
@@ -697,20 +761,29 @@ def _multiplicative_update_h(X, W, H, beta_loss, l1_reg_H, l2_reg_H, gamma):
         denominator = denominator + l2_reg_H * H
     denominator[denominator == 0] = EPSILON
 
+    if A is not None and B is not None and slice_index > 0:
```

Comment:
Why is slice_index > 0 necessary ? you don't want to update after the first batch ?

Reply:
I was imposing NMF and MiniBatchNMF giving the same result when `batch_size==n_samples` without passing `batch_size` neither `n_samples` to `_multiplicative_update_h` ... I'm wondering right now if this is the right solution...

Reply:
I don't know if we want to impose that. Although it makes it easier to test :) 

If we initialize A and B to zeros, it should give the same results as no minibatch, right ? We could only test that when initializing A and B to zeros we have equivalence, but I'm not sure.

Otherwise I think it would be easier to understand if you pass a bool `is_first_minibatch` or something like that instead of slice_index, with adding a comment why we ignore the first minibatch.

Reply:
I have introduced a new boolean parameter `single_batch`: it is `True` for `batch_size >= n_samples`. This way the first iteration updates if there are other batches waiting in the queue.

Reply:
@jeremiedbb I'm resolving this conversation as the code is now outdated: I'll let you review the new version.